optuna/integration/dask.py

```diff
@@ -27,6 +27,8 @@
 
 with try_import() as _imports:
     import distributed
+    from distributed.protocol.pickle import dumps
+    from distributed.protocol.pickle import loads
```

Comment:
`RDBStorages` uses JSON to serialize `user_attrs` and `system_attrs` actually.
https://github.com/optuna/optuna/blob/32e2ac2ecda89cac714f248b39c18cb3063768e3/optuna/storages/_rdb/storage.py#L762-L781

Although it might be a bit controversial since users can set any Python object via `trial.set_user_attr(key: str, value: Any)`, we do not restrict the type of `user_attrs` and `system_attrs` for `InMemoryStorage`.

Reply:
Good to know. The fact that these attributes could be any Python object is what motivated me to use `pickle` for serialization here.